enhancements/network/user-defined-network-segmentation.md

```diff
@@ -0,0 +1,956 @@
+service exists. Services that exist before the user-defined network is assigned to a namespace will result in
+OVN-Kubernetes executing a re-sync on all services in that namespace, and updating all load balancers. Keep in mind that
+pods must not exist in the namespace when the namespace is assigned to a new network or the new network assignment will
+not be accepted by OVN-Kubernetes.
+
+Services in a user-defined network will be reachable by other namespaces that share the same network.
+
+As previously mentioned, Kubernetes API and DNS services will be accessible by all pods.
+
+Endpoint slices will provide the IPs of the cluster default network in Kubernetes API. For this implementation the required
+endpoints are those IP addresses which reside on the user-defined primary network. In order to solve this problem,
+OVN-Kubernetes may create its own endpoint slices or may choose to do dynamic lookups at runtime to map endpoints to
+their primary IP address. Leveraging a second set of endpoint slices will be the preferred method, as it creates less
+indirection and gives explicit Kube API access to what IP addresses are being used by OVN-Kubernetes.
+
+Kubelet health checks to pods are queried via the cluster default network. When endpoints are considered unhealthy they
+will be removed from the endpoint slice, and thus their primary IP will be removed from the OVN load balancer. However,
+it is important to note that the healthcheck is being performed via the cluster default network interface on the pod, 
+which ensures the application is alive, but does not confirm network connectivity of the primary interface. Therefore,
+there could be a situation where OVN networking on the primary interface is broken, but the default interface continues
+to work and reports 200 OK to Kubelet, thus rendering the pod serving in the endpoint slice, but unable to function.
+Although this is an unlikely scenario, it is good to document.
+
+### Network Policy
+
+Configuring network policies that allow traffic between user-defined network namespaces will not be invalidated or
+ignored by OVN-Kubernetes. If a user configures a network policy that would have no effect, (like stopping namespace A
+from talking to B on different networks) then OVN-Kubernetes will still configure this policy in OVN. These policies
+will also have no effect between namespaces that are assigned to different networks as there will be no network
+connectivity between the namespaces. However, there may still be a desire to configure network policy between namespaces
+that share the same network, as well as for other reasons like ingress/egress control. Additionally, policies that
+block/allow cluster egress or ingress traffic will still be enforced.
+
+### API Extensions
+
+The main API extension here will be a namespace scoped network CRD as well a cluster scoped network CRD. These CRDs
```

Comment:
```suggestion
The main OpenShift API extension here will be a namespace scoped network CRD as well a cluster scoped network CRD. These CRDs
```

(because there's also the ovn-kubernetes-internal addition of `primaryNetwork` to the CNI config)

Reply:
hmm, but this isnt OpenShift API. The network CRD will belong to k8s.ovn.org. The NAD configuration is free form.

Reply:
Ah, OK. I think this section of the enhancement template is mostly intended to be "stuff that will touch `openshift/api`". But I guess talking about all the APIs its adding everywhere is good.